R/nm_to_regimen.R

```diff
@@ -19,29 +23,44 @@ nm_to_regimen <- function(data,
   if(! "time" %in% colnames(data)) {
     stop("TIME column is required in source dataset!")
   }
-  m <- match(c("id", "mdv", "evid", "amt", "time", "rate"), colnames(data), 0)
+  m <- match(c("id", "mdv", "evid", "amt", "time", "rate", "cmt"), colnames(data), 0)
   m <- m[m>0]
   data <- data[,m]
   doses <- data[data$evid == 1,]
-  dum <- data[data$evid == 2,]
   ids <- unique(doses$id)
   if(first_only) {
     ids <- ids[1]
   }
-  type <- "bolus"
-  if("rate" %in% colnames(doses) &! 0 %in% doses$rate) {
-    type <- "infusion"
-  }
   reg <- list()
   for(i in 1:length(ids)) {
     tmp <- doses[doses$id == ids[i],]
     if(reset_time) {
       tmp$time <- tmp$time - min(tmp$time)
     }
-    if(type == "infusion") {
-      reg[[i]] <- new_regimen(amt = tmp$amt, times = tmp$time, type = "infusion", t_inf = tmp$amt / tmp$rate)
+    if (!is.null(dose_cmts)){
+      # map cmt to dose type
+      reg[[i]] <- new_regimen(
+        amt = tmp$amt,
+        times = tmp$time,
+        cmt = tmp$cmt,
```

Comment:
I think we could skip this argument. if it is null, `PKPDsim::sim` will use the compartment definitions of the model file, which is probably safer.

Reply:
Could you help me understand this case with an example? It is possible that NONMEM-style data and PKPDsim model definitions use different compartments for one dose type.

Reply:
https://github.com/InsightRX/mipdeval/pull/25/files

This is the **only** relevant place this function is currently being used, where we want to specify which compartments are from which doses using the NONMEM data set. Leaving it to be assumed by the model downstream, especially when CMT should be specified in the data set anyway, is not optimal imo.